### DISCUSSION TITLE
Erro na hora de salvar o arquivo

### DISCUSSION THREAD

**Discussion by Roger-Aguiar** [balta-io/forum#79] [Tira Dúvidas] [answered]
@andrebaltieri , eu fiz o código de salvar conforme o seu porém ele está engolindo a primeira letra mesmo depois que salva, no seu não está acontecendo isso, or que pode ser isso? Segue meu código: 
`public class FileCreator
{
    public string? Text { get; set; }
    Menu menu = new Menu();

    public void CreateFile()
    {        
        Console.WriteLine("Enter your text and press ESC to save the file.\n");

        do
        {
            Text += Console.ReadLine();
            Text += Environment.NewLine;//It allows the user types multiple lines.
        } while (Console.ReadKey().Key != ConsoleKey.Escape);
        //ConsoleKey.Escape = "ESC" key
        Save();
        menu.RunMenu();
    }
    
    private void Save()
    {    
        Console.Clear();

        Console.Write("Enter the path that you want to save this file: ");
        var path = Console.ReadLine();

        using(var file = new StreamWriter(path))
        {
            file.Write(Text);
        }

        Console.WriteLine($"The file {path} was saved successfully!");
        Console.WriteLine("Press any key to go back to menu...");
        Console.ReadLine();
    }
}`

Veja como ficou meu arquivo:
Roger
ana
obson

Deveria ser:

Roger
Vana
Robson

Alguma dica por gentileza?

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @Roger-Aguiar tem a explicação aqui 👇
https://github.com/balta-io/2801/discussions/11
